### PR TITLE
APIv2 Add network list filtering

### PR DESCRIPTION
**Pull request by Luap99**
Add the filter option to the libpod endpoint.
Add support for the name filter on the docker endpoint.

Add apiv2 tests for the network list endpoints.
Enable podman network integration tests for remote.

### CODE REVIEW

**Comment by Luap99** [comment]
Ref #7462

**Comment by rhatdan** [comment]
@baude @jwhonce @mheon PTAL

**Comment by QiWang19** [comment]
LGTM

File: pkg/api/server/register_networks.go [hunk 1]
```diff
@@ -106,7 +111,7 @@ func (s *APIServer) registerNetworkHandlers(r *mux.Router) error {
 	//    required: true
 	//    description: the name of the network
 	//  - in: query
-	//    name: Force
+	//    name: force
```
Comment (vrothberg) [inline pkg/api/server/register_networks.go:114]: Nit (for cleaner potential cherry-picking): Can you break this change out into a separate commit?

**Review by vrothberg — COMMENTED** [review]
Other than the nit, LGTM.

But I want a head nod from @baude and/or @jwhonce

File: pkg/api/server/register_networks.go (hunk 1 above)
Comment (mheon) [reply pkg/api/server/register_networks.go:114]: I wouldn't worry. We're past the point that anything should be going into 2.0 without a critical bug being behind it.

**Review by vrothberg — COMMENTED** [review]
LGTM, given tests are passing I am good to merge

@mheon @rhatdan PTAL
